File: facke-iot-service/src/facke-iot/domain/production.dto.ts
import { Time } from "./time.enum";

export class ProductionDto {

  private id: number;

  private time: Time;

  private product: string;

  private goodPiece: number;

  private badPiece: number;

  private total: string;

  constructor(productionDto: ProductionDto) {
    this.id = productionDto.id;
    this.time = productionDto.time;
    this.product = productionDto.product;
    this.goodPiece = productionDto.goodPiece;
    this.badPiece = productionDto.badPiece;
    this.total = productionDto.total;
  }
}

File: iot-hub-api/src/main/java/com/soft/iot_hub_api/dto/ProductionDTO.java
package com.soft.iot_hub_api.dto;

import com.soft.iot_hub_api.domain.enums.Time;
import lombok.*;

import java.io.Serializable;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ProductionDTO implements Serializable {

    private Long id;

    private Time time;

    private String product;

    private Integer goodPiece = 0;

    private Integer badPiece = 0;

    @Getter(AccessLevel.NONE)
    private Integer total;

    public Integer getTotal() {
        return goodPiece + badPiece;
    }
}
